### ISSUE TITLE
Example Shell throws exception + No plug-in alternative

### ISSUE THREAD

**Issue by thumbtech** [WyriHaximus/FlyPie#10]
While I appreciate these efforts, with no CakePHP 3.x code sample provided (and it is *not* totally intuitive) I wonder whether anybody is actually using this plug-in? ... With CakePHP 3.5.3 & flysystem 1.0.41 the plug-in is not working as advertised. The idea is nice, but it may be better to just skip the middleman (see last code sample).

Anyway, some code samples:

In config:
```php
...
    'WyriHaximus' => [
        'FlyPie' => [
            'lpsaa' => [
                'adapter' => 'Sftp',
                'vars' => [
                    'host' => 'ftp.host.com',
                    'port' => 22,
                    'username' => 'user',
                    'password' => 'password',
                    'root' => '/path',
                    'timeout' => 10,
                ],
            ],
        ],
    ],
...
```

A shell:
```php
namespace App\Shell;

use Cake\Console\Shell;
use WyriHaximus\FlyPie\FilesystemsTrait;

class FlypieTestShell extends Shell
{
	use FilesystemsTrait;

	public function main()
	{
		$this->out('Flypie Test:');
		$contents = $this->filesystem('lpsaa')->listContents();
		print_r($contents);
	}
}
```

Output:
```
C:\app>bin\cake flypie_test
Flypie Test:
Exception: Argument 1 passed to League\Flysystem\Adapter\AbstractFtpAdapter::__construct() must be of the type array, string given in [C:\app\vendor\league\flysystem\src\Adapter\AbstractFtpAdapter.php, line 89]
2017-10-13 13:19:05 Error: [TypeError] Argument 1 passed to League\Flysystem\Adapter\AbstractFtpAdapter::__construct() must be of the type array, string given in C:\app\vendor\league\flysystem\src\Adapter\AbstractFtpAdapter.php on line 89
Stack Trace:
#0 [internal function]: League\Flysystem\Adapter\AbstractFtpAdapter->__construct('ftp.host.com', 22, 'user', 'password', '/path', 10)
#1 C:\app\vendor\wyrihaximus\fly-pie\src\FilesystemRegistry.php(170): ReflectionClass->newInstanceArgs(Array)
#2 C:\app\vendor\wyrihaximus\fly-pie\src\FilesystemRegistry.php(79): WyriHaximus\FlyPie\FilesystemRegistry::adapter('Sftp', Array)
#3 C:\app\vendor\wyrihaximus\fly-pie\src\FilesystemRegistry.php(38): WyriHaximus\FlyPie\FilesystemRegistry::create('lpsaa')
#4 C:\app\vendor\wyrihaximus\fly-pie\src\FilesystemsTrait.php(18): WyriHaximus\FlyPie\FilesystemRegistry::retrieve('lpsaa')
#5 C:\app\src\Shell\FlypieTestShell.php(15): App\Shell\FlypieTestShell->filesystem('lpsaa')
#6 C:\app\vendor\cakephp\cakephp\src\Console\Shell.php(508): App\Shell\FlypieTestShell->main()
#7 C:\app\vendor\cakephp\cakephp\src\Console\CommandRunner.php(141): Cake\Console\Shell->runCommand(Array, true)
#8 C:\app\bin\cake.php(12): Cake\Console\CommandRunner->run(Array)
```

Hacking config thusly tricks FilesystemRegistry and works:
```php
...
    'WyriHaximus' => [
        'FlyPie' => [
            'lpsaa' => [
                'adapter' => 'Sftp',
                'vars' => [
                    'params' => [
                        'host' => 'ftp.host.com',
                        'port' => 22,
                        'username' => 'user',
                        'password' => 'password',
                        'root' => '/path',
                        'timeout' => 10,
                    ],
                ],
            ],
        ],
    ],
...
``` 

FWIW essentially same functionality without plug-in:
```php
namespace App\Shell;

use Cake\Console\Shell;
use Cake\Core\Configure;

use League\Flysystem\Filesystem;
use League\Flysystem\Sftp\SftpAdapter;

class FlypieTestShell extends Shell
{
	public function main()
	{
		$this->out('Flypie Test:');
		// using the original config example
		$filesystem = new Filesystem(new SftpAdapter(Configure::read('WyriHaximus.FlyPie.lpsaa.vars')));
		$contents = $filesystem->listContents();
		print_r($contents);
	}
}
```

**Comment by WyriHaximus** [comment]
Hey @thumbtech thank you for your feedback. I've created #11 to address the issues you've raised. There are some minor things that can be improved in the docs but the easiest way to fix your issue is to use:

```
    'WyriHaximus' => [
        'FlyPie' => [
            'lpsaa' => [
                'adapter' => 'Sftp',
                'vars' => [
                    [
                        'host' => 'ftp.host.com',
                        'port' => 22,
                        'username' => 'user',
                        'password' => 'password',
                        'root' => '/path',
                        'timeout' => 10,
                    ],
                ],
            ],
        ],
    ],
```

**Comment by WyriHaximus** [comment]
@thumbtech I've addressed your issues in #11, care to take a look before I merge it?
